workflow/controller/operator.go

```diff
@@ -1880,23 +1897,24 @@ func (woc *wfOperationCtx) initializeNodeOrMarkError(node *wfv1.NodeStatus, node
 }
 
 // Creates a node status that's completely initialized and marked as finished
-func (woc *wfOperationCtx) initializeCacheNode(nodeName string, resolvedTmpl *wfv1.Template, templateScope string, orgTmpl wfv1.TemplateReferenceHolder, boundaryID string, outputs *wfv1.Outputs, memStat *wfv1.MemoizationStatus, messages ...string) *wfv1.NodeStatus {
+func (woc *wfOperationCtx) initializeCacheNode(nodeName string, resolvedTmpl *wfv1.Template, templateScope string, orgTmpl wfv1.TemplateReferenceHolder, boundaryID string, memStat *wfv1.MemoizationStatus, messages ...string) *wfv1.NodeStatus {
 	if resolvedTmpl.Memoize == nil {
 		err := fmt.Errorf("Cannot initialize a cached node from a non-memoized template")
 		woc.log.WithFields(log.Fields{"namespace": woc.wf.Namespace, "wfName": woc.wf.Name}).WithError(err)
 		panic(err)
 	}
 	woc.log.Debug("Initializing cached node ", nodeName, common.GetTemplateHolderString(orgTmpl), boundaryID)
-	nodePhase := wfv1.NodePending
-	if memStat.Hit {
-		nodePhase = wfv1.NodeSucceeded
-	}
-	node := woc.initializeExecutableNode(nodeName, wfutil.GetNodeType(resolvedTmpl), templateScope, resolvedTmpl, orgTmpl, boundaryID, nodePhase, messages...)
+	node := woc.initializeExecutableNode(nodeName, wfutil.GetNodeType(resolvedTmpl), templateScope, resolvedTmpl, orgTmpl, boundaryID, wfv1.NodePending, messages...)
```

Comment:
I actually prefer the two functions, since it makes the part of the code very clear:
<img width="808" alt="image" src="https://user-images.githubusercontent.com/614838/94450143-c09e7e00-0161-11eb-9037-bce8d9e1deeb.png">

Reply:
I liked the separate function but my only concern is code has too many `initializeXXXXXNode`.  `initializeCacheNode` is just calling initializeExecutableNode with check the memorization.

Reply:
I see what you mean now, will remove this function

Reply:
The function actually also adds the `memoizationStatus` to the node, and I would like to keep it to keep the legibility of the code above